src/hooks/useDatePicker.ts

```diff
@@ -191,7 +191,7 @@ export const useDatePicker: UseDatePickerHookType = ({
                 dateValue: formatDate(weekDayZdt, undefined, format),
                 label: localisationHelpers.localiseWeekLabel(
                     weekDayZdt.withCalendar(localeOptions.calendar),
-                    localeOptions
+                    {...localeOptions, calendar: resolvedOptions.calendar}
```

Comment:
this `calendar: resolvedOptions.calendar` should already be part of the `localeOptions` already (if it's about the type, you can cast up when it's set)

Reply:
resolvedOptions has calendar as `String` but localOptions has calendar as `Object` with `id` property. I can't cast it as I am using `id` inside useNavigation